azkaban-exec-server/src/main/java/azkaban/execapp/ProjectCacheCleaner.java

```diff
@@ -243,16 +249,19 @@ private long bytesToMB(final long bytes) {
    * This method will block until all active cleanup threads finish deleting submitted
    * cleanup jobs.
    */
-  public void finishPendingCleanup() {
+  @VisibleForTesting
+  void finishPendingCleanup() {
     final long start = System.currentTimeMillis();
     try {
```

Comment:
I think this whole logic can be written inside one single try catch finally like this,

try {
        barrier.lock();
        while (!projectsUnderDeletion.isEmpty()) {
          log.info("{} entries left in the cache directory deletion Q. Waiting for the cleanup to finish",
              this.projectsUnderDeletion.size());
          this.emptyQCond.await(10, TimeUnit.SECONDS);
        }
      log.info("Took {} ms to complete ongoing cache cleanup.", (System.currentTimeMillis() - start));
    } catch (InterruptedException e) {
      e.printStackTrace();
    } finally {
      this.barrier.unlock();
    }